### PR TITLE
Upgrading the programming language doc

### PR DESCRIPTION
**Pull request by jabrena**
Upgrading doc about the programming languages to add a section about Java.

Juan Antonio

### CODE REVIEW

File: docs/programming-languages.md [hunk 1]
```diff
@@ -50,6 +50,17 @@ to learn.
 
 [Get started with JavaScript](https://github.com/wasabifan/ev3dev-lang-js){: .btn .btn-default }
 
+## Java
+<http://ev3dev-lang-java.github.io/>
+
+Java is a general-purpose computer programming language that is concurrent,
+
```
Comment (dlech) [inline docs/programming-languages.md]: The blank lines cause this to be 3 separate paragraphs.

https://github.com/ev3dev-lang-java/ev3dev.github.io/blob/c528b13073a600fa508eee160eef9434c7739931/docs/programming-languages.md#java

Please fix.

**Comment by WasabiFan** [comment]
We need to make sure we keep things fair between this library and @mob41's. The distinction this page previously made was that one followed the LeJOS model and the other followed an ev3dev-lang format; I think that distinction still seems reasonable. See here for reference, and note that @mob41 has been updating it recently: https://github.com/mob41/ev3dev-lang-java

**Comment by jabrena** [comment]
Hi @WasabiFan, 

Oki. I have just updated with the info. :)

File: docs/programming-languages.md (hunk 1 above)
Comment (jabrena) [inline docs/programming-languages.md]: Fixed.

**Comment by jabrena** [comment]
Good morning @dlech , @WasabiFan, 

Yesterday, I updated the document with the changes that you said.
Can you review the changes in the doc, please?

Cheers

Juan Antonio

**Comment by mob41** [comment]
@WasabiFan As @jabrena 's library does not follows the ```ev3dev-lang``` unified language model, instead, interfacing LeJOS, having two libraries with the same name but different functions will bring a confusion to the users. It is recommended to have his library named ```ev3dev-compat-lejos```, which represents a compatibility layer to LeJOS.

File: docs/programming-languages.md [hunk 2]
```diff
@@ -50,6 +50,17 @@ to learn.
...
 [Get started with JavaScript](https://github.com/wasabifan/ev3dev-lang-js){: .btn .btn-default }
 
+## Java with LeJOS.
+<http://ev3dev-lang-java.github.io/>
+
+Java is a general-purpose computer programming language that is concurrent,class-based, object-oriented. In Java 8, the language added some functional features very interesting to learn.
+
+The project has support for EV3 Sensors and Actuators. The project includes the support for LeJOS local navigation stack.
+
+In this project, the Objects included in the different libraries follow the LeJOS Interfaces, not confuse with another Java project based on [ev3dev unified language binding for Java](https://github.com/mob41/ev3dev-lang-java)
```
Comment (WasabiFan) [inline docs/programming-languages.md]: "not confuse with" -> "not to be confused with"

File: docs/programming-languages.md [hunk 3]
```diff
@@ -50,6 +50,17 @@ to learn.
 
 [Get started with JavaScript](https://github.com/wasabifan/ev3dev-lang-js){: .btn .btn-default }
 
+## Java with LeJOS.
+<http://ev3dev-lang-java.github.io/>
+
+Java is a general-purpose computer programming language that is concurrent,class-based, object-oriented. In Java 8, the language added some functional features very interesting to learn.
```
Comment (WasabiFan) [inline docs/programming-languages.md]: Space after comma

Comment (WasabiFan) [inline docs/programming-languages.md]: "and" instead of comma for last item in list

File: docs/programming-languages.md [hunk 4]
```diff
@@ -50,6 +50,17 @@ to learn.
 
 [Get started with JavaScript](https://github.com/wasabifan/ev3dev-lang-js){: .btn .btn-default }
 
+## Java with LeJOS.
```
Comment (WasabiFan) [inline docs/programming-languages.md]: Remove period

Comment (WasabiFan) [inline docs/programming-languages.md]: "with" makes it sound like LeJOS is actually running. Could you say " Java (LeJOS-style)" or "Java (LeJOS-compatible)"?

**Comment by WasabiFan** [comment]
I agree with @mob41, in that the two libraries should be unique in name and that the LeJOS one would be the logical one to be renamed. I've generally attempted to stay away from having to deal with that, but if we are going to put these in the main language list we should sort it out.

**Comment by jabrena** [comment]
Hi @WasabiFan, @mob41 It is not a bad idea. In the beginning of the project I tried to use the binding features but later I considered that it could be complicated. In this iteration, I will change the name of the project and it is a big change and I need to do many changes. Today, I will do the changes to upgrade the programming language doc.

Juan Antonio

**Comment by jabrena** [comment]
Hi @WasabiFan,

I upgraded the doc with the notes.

**Comment by WasabiFan** [comment]
Sounds good to me!

It seemed like you were planning to make such a release fairly imminently; if that's the case, I think that we should hold off on merging this until you get the rename completed.

**Comment by jabrena** [comment]
I would like to do in the next release `v0.7.0` not in this one.

For this iteration, I think that we have clarified both projects without any problem in the doc.
In the next iteration with calm, I will change the project name. The most important thing is the  communicate that the library has a good support with latest ev3dev kernel. Later, the users could review the details.

Block this merge request after the clarification in the doc is not necessary, I think in my humble opinion.

Are you ok @WasabiFan ?

Cheers

Juan Antonio

**Comment by jabrena** [comment]
Many thanks @WasabiFan, @dlech.

Issue created to rename the project.
https://github.com/ev3dev-lang-java/ev3dev-lang-java/issues/311

Cheers

Juan Antonio

**Comment by WasabiFan** [comment]
@mob41 feel free to open a PR to move your Java library into this section as well. These two were originally put in the section below because they hadn't been updated for a while and were incompatible with breaking changes, but now that they are healthy they'd be fine as full entries.

**Pull request merged**